JASP-Desktop/analysis/analyses.cpp

```diff
@@ -248,8 +248,8 @@ Json::Value Analyses::asJson() const
 	Json::Value analysesJson		= Json::objectValue,
 				analysesDataList	= Json::arrayValue;;
 
-	for(auto idAnalysis : _analysisMap)
-			analysesDataList.append(idAnalysis.second->asJSON());
+	applyToAll([&analysesDataList](const Analysis * analysis)
```

Comment:
@boutinb here is the actual fix, the rest is just rearranging the deckchairs on the titanic. 
But the problem was that using an iterator on a std::map does not give the proper order at all. So the analyses weren't stored in the right order in the jasp-file. Which is why order was lost.

The function applyToAll was specifically added to make sure all analyses are processed in the right order. Why we didn't do it here, I don't know.